docs/controllers.md

```diff
@@ -0,0 +1,152 @@
+2. The cross-chain protocol specific information.
+3. Passthrough metadata that has to be attached to the cross-chain transfer.
+
+The cross-chain protocol specific information is encoded with the `Attributes` field with an any
+type that implements the `ForwardingAttributes` interface. For additional details on the
+implementation, please refer to the
+[proto definition](https://github.com/noble-assets/orbiter/blob/main/proto/noble/orbiter/core/v1/orbiter.proto#L35-L55)
+
+Forwarding controllers are elements of the Orbiter module used to interpret the protocol-specific
+information and use them to forward funds. Based on the selected bridge, we can have:
+
+- **AutoCCTP**: Automatic forwarding of funds via the Cross-Chain Transfer Protocol.
+- **AutoLane**: Automatic forwarding of funds via the Hyperlane protocol.
+- **AutoIBC**: Automatic forwarding of funds via the Inter-BlockChain protocol.
+
+## Hyperlane
+
+The Hyperlane protocol is a mailbox-based implementation of a permissionless bridge. Mailboxes are
+the entrypoints used to send and receive messages via Hyperlane. The cross-chain transfer is built
+around the general message passing functionality, Hyperlane **Core**, as an application called
+**Warp**. The architecture is similar to how IBC core and ICS20 are structured. When we send coins,
+we first call into the Warp server which then calls into the Core, and when we receive them, the
+flow is in the opposite direction. Two elements are important to be aware of:
+
+- Interchain Security Modules (ISMs): these modules are used to verify a received message.
+- Post dispatch hooks: these modules are used to execute post dispatch logic when a message is sent.
+
+For additional information, please refer to the [Hyperlane docs](https://docs.hyperlane.xyz/).
+
+The Hyperlane controller is the specific implementation of the controller interface designed to
+handle Hyperlane transfers. A Hyperlane transfer is defined by the
+[`HypAttributes`](https://github.com/noble-assets/orbiter/blob/main/proto/noble/orbiter/controller/forwarding/v1/hyperlane.proto#L12-L48)
+type, which is one of the concrete implementations of the forwarding attributes interface. Below is
+reported the proto implementation of the attributes:
+
+```protobuf
+message HypAttributes {
+  option (cosmos_proto.implements_interface) = "orbiter.core.v1.ForwardingAttributes";
+
+  bytes token_id = 1;
+  uint32 destination_domain = 2;
+  bytes recipient = 3;
+  bytes custom_hook_id = 4;
+  string gas_limit = 5 [
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true,
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int"
+  ];
+  cosmos.base.v1beta1.Coin max_fee = 6 [
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true,
+    (amino.encoding) = "legacy_coins",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
+  ];
+  string custom_hook_metadata = 7;
+}
+```
```

Comment:
not sure we should copy paste the whole thing here instead of just pointing to the GitHub link.